src/components/CommenAdder.jsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect, } from "react";
+import { getUsers, postComment } from "../utils/api";
+import { useParams } from "react-router-dom";
+
+
+
+function CommentAdder (setComments) {
+    const {id} = useParams();
+    const [users, setUsers] = useState([]);
+    const [addComment, setAddComment] = useState("");
+    const [selectedUser, setSelectedUser] = useState("");
+
+
+    useEffect(() => {
+        getUsers()
+        .then((response) => {
+            setUsers(response.users)
+        })
+        .catch((err) => {
+            console.log(err)
+        })
+    }, []);
+
+    function handleAddComment(event) {
+        setAddComment(event.target.value);
+      }
+
+      function handleUserChange(event) {
+        setSelectedUser(event.target.value);
+      }
+
+    
+      function handleSubmit(event) {
+        event.preventDefault();
+    
+        postComment(id, selectedUser, addComment)
+          .then((response) => {
+            setComments((currentComments) => {
+                return [response, ...currentComments]
+            })
+        })
+        .catch((err) => {
+            console.log(err)
+        });
+        setSelectedUser("");
+        setAddComment("");
+        }
+
+
```

Comment:
On the whole a nice implementation of a controlled component well done 🙌 
However, as you say the comment you add doesn't appear until you refresh.
This is because you are successfully making the post request, but you aren't successfully updating the state of the list of comments on lines 38-40.
If you have a look at your console log after posting a message, line 48 is logging an error which may help you spot the reason behind this, it doesn't look like `setComments` function has been brought in via props properly.